task_1/anna.zinovatnaya/src/main/java/task_1/Dialog.java

```diff
@@ -0,0 +1,272 @@
+
+    System.out.println("Type 'read not committed elements' to read all not committed elements");
+    System.out.println("Type 'read committed elements' to read all committed elements");
+
+    System.out.println("Type 'exit' to exit");
+
+
+    boolean endOfDialog = false;
+
+    while (endOfDialog == false) {
+
+      System.out.print("> ");
+
+      while (!scanner.hasNextLine()) {}
+
+      option = scanner.nextLine();
+
+      switch (option) {
+        case "new file":
+          this.addFile();
+          break;
+        case "new folder":
+          this.addFolder();
+          break;
+        case "current folder":
+          this.getCurrentFolder();
+          break;
+        case "change folder":
+          this.changeFolder();
+          break;
+        case "root folder":
+          this.changeFolderToRoot();
+          break;
+        case "commit":
+          this.commit();
+          break;
+        case "log":
+          this.log();
+          break;
+        case "read not committed element":
+          this.readElement();
+          break;
+        case "read committed element":
+          this.readCommittedElement();
+          break;
+        case "read not committed elements":
+          this.readElements();
+          break;
+        case "read committed elements":
+          this.readCommittedElements();
+          break;
+        case "exit":
+          endOfDialog = true;
+          break;
+        default:
+          System.out.println("Incorrect input!");
+      }
+    }
+  }
+
+
+
+  void addFile() {
+
+    Scanner scanner = new Scanner(System.in, "UTF-8");
+    System.out.println("Type your text:");
+
+    String temp = scanner.nextLine();
+
+    Element file = new File(temp);
+
+    this.currentFolder.addFile((File) file);
+  }
+
+
+
+  void addFolder() {
+
+    Element tree = new Folder();
+
+    this.currentFolder.addFolder((Folder) tree);
+
+    System.out.println("New folder created");
+  }
+
+
+
+  void getCurrentFolder() {
+    System.out.println("Current folder " + currentFolder.getHashCode());
+  }
+
+
+
+  void changeFolder() {
+    Scanner scanner = new Scanner(System.in, "UTF-8");
+    System.out.println("Type folder hash code:");
+    String input = scanner.nextLine();
+
+
+    boolean folderExists = rootFolder.containsFolder(input);
+
+    if (folderExists) {
+      currentFolder = rootFolder.getFolder(input);
+      System.out.println("Current folder was changed to " + input);
+    } else {
+      System.out.println("No such folder");
+    }
+
+  }
+
+
+
+  void changeFolderToRoot() {
+
+    currentFolder = rootFolder;
+    System.out.println("Current folder was changed to root folder");
+
+  }
+
+
+
+  void commit() {
+    Commit commit;
+
+    Scanner scanner = new Scanner(System.in, "UTF-8");
+    System.out.println("Type commit message:");
+    String message = scanner.nextLine();
+
+
+    if (commits.isEmpty()) {
```

Comment:
You can move it into separate method, so it would look something like Commit commit = someMethod(...); This way code is more readable.